### PR TITLE
fix(MAASENG-1313): Remove duplicated tab navigation items

### PR DESCRIPTION
**Pull request by ndv99**
## Done

- Removed duplicated tab navigation items on KVM pages
- Updated KVM header to display "LXD" or "Virsh" instead of "KVM"

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to `/kvm/lxd`
- Ensure the title is "LXD" (window and page header)
- Ensure the old tab links are not rendered
- Navigate to `/kvm/virsh`
- Ensure the title is "Virsh" (window and page header)
- Ensure the old tab links are not rendered

## Fixes

Fixes [MAASENG-1313](https://warthogs.atlassian.net/browse/MAASENG-1313?atlOrigin=eyJpIjoiNzExYTAxMWY0MjMyNGUyMTk0NGNjZTkxMTBlZjczYzMiLCJwIjoiaiJ9)

### Before
image

### After
image
****

[MAASENG-1313]: https://warthogs.atlassian.net/browse/MAASENG-1313?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx [hunk 1]
```diff
@@ -18,21 +18,21 @@ import { getFormTitle } from "app/kvm/utils";
 import { actions as podActions } from "app/store/pod";
 import podSelectors from "app/store/pod/selectors";
 
-type Props = {
+type Props = SectionHeaderProps & {
```
Comment (petermakowski) [inline src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx:21]:
```suggestion
type Props = Required<Pick<SectionHeaderProps, "title">> & {
```

File: src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx [hunk 2]
```diff
@@ -67,23 +67,7 @@ const KVMListHeader = ({
...
-        {
-          active: virshTabActive,
-          component: Link,
-          "data-testid": "virsh-tab",
-          label: "Virsh",
-          to: urls.kvm.virsh.index,
-        },
-      ]}
-      title="KVM"
+      title={"title" in props && !!props.title ? props.title : "KVM"}
```
Comment (petermakowski) [inline src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx:70]:
```suggestion
      title={title}
```

File: src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx [hunk 3]
```diff
@@ -18,21 +18,21 @@ import { getFormTitle } from "app/kvm/utils";
...
-type Props = {
+type Props = SectionHeaderProps & {
   sidePanelContent: KVMSidePanelContent | null;
   setSidePanelContent: KVMSetSidePanelContent;
 };
 
 const KVMListHeader = ({
   sidePanelContent,
   setSidePanelContent,
+  ...props
```
Comment (petermakowski) [inline src/app/kvm/views/KVMList/KVMListHeader/KVMListHeader.tsx:29]:
```suggestion
  title,
```

File: src/app/kvm/views/KVMList/KVMList.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ const KVMList = (): JSX.Element => {
   const hasVirsh = virshKvms.length > 0;
   const showingLXD = location.pathname.endsWith(urls.kvm.lxd.index);
   const showingVirsh = location.pathname.endsWith(urls.kvm.virsh.index);
-  useWindowTitle("KVM");
+  useWindowTitle(showingLXD ? "LXD" : "Virsh");
```
Comment (petermakowski) [inline src/app/kvm/views/KVMList/KVMList.tsx:42]: The way this condition works means KVM title will never be rendered as Virsh is always going to be the fallback value.

```suggestion
  const title = showingLXD ? "LXD" : showingVirsh ? "Virsh" : "KVM";
  useWindowTitle(showingLXD ? "LXD" : "Virsh");
```

File: src/app/kvm/views/KVMList/KVMList.tsx [hunk 5]
```diff
@@ -95,6 +95,7 @@ const KVMList = (): JSX.Element => {
         <KVMListHeader
           setSidePanelContent={setSidePanelContent}
           sidePanelContent={sidePanelContent}
+          title={showingLXD ? "LXD" : "Virsh"}
```
Comment (petermakowski) [inline src/app/kvm/views/KVMList/KVMList.tsx:98]:
```suggestion
          title={title}
```

**Comment by petermakowski** [comment]
@ndv99 This PR does not include duplicated items on machines page which was part of this task, but feel free to update in a separate PR if you prefer.

**Comment by ndv99** [comment]
> @ndv99 This PR does not include duplicated items on machines page which was part of this task, but feel free to update in a separate PR if you prefer.

I should have mentioned that in the description - Amy isn't currently sure how she wants to handle the items at the top of the navigation (Images, Pools, AZs etc.), and one possible solution is just taking AZs, Pools, and Tags out of the nav and leaving these tabs in place. She'll get back to us at the end of the week on it but the ones removed in this PR can definitely stay gone

**Pull request merged**